### DISCUSSION TITLE
Support for flashing the Zephyr based application on i.MX RT 1160 EVK through SDP Mode(USB-HID/ UART) & PyOCD runner

### DISCUSSION THREAD

**Discussion by vishnusudhan** [pyocd/pyOCD#1410] [General]
Hi,

I am very new to this platform, I am using i.MX RT 1166 EVK board for my development purpose. I could able to build & flash the zephyr based application using west flash command using JTAG (JLink segger) alone.

The i.MX RT1166 EVK has the SDP mode flash support through USB-HID / UART, Is there any support / reference available to flash the zephyr based application through SDP mode (USB-HID / UART).

I could not able to flash the zephyr application for i.MX RT1160 EVK platform through PYOCD , seems like PyOCD runner support for i.MX RT1160 not yet added. Suggest any steps / updated zephyr source code to flash the application using PyOCD runner for i.MX RT1160EVK.

**Comment by flit** [comment]
(Apologies for taking a long time to reply! 🙇🏽)

PyOCD doesn't support SDP mode because it is a debug tool rather than bootloader interface. NXP has tools for SDP mode.

Please see #1411 for info about an issue with the RT1160 board ID (an invalid referenced target type). Please see the [target support docs](https://pyocd.io/docs/target_support.html) for how to add the RT1160 via CMSIS-Packs. You may also try the `mimxrt1170_cm7` built-in target.
